py-polars/tests/unit/conftest.py

```diff
@@ -138,3 +141,53 @@ def iso8601_tz_aware_format_datetime(request: pytest.FixtureRequest) -> list[str
 @pytest.fixture(params=ISO8601_FORMATS_DATE)
 def iso8601_format_date(request: pytest.FixtureRequest) -> list[str]:
     return cast(List[str], request.param)
+
+
+class MemoryUsage:
+    """
+    Provide an API for measuring peak memory usage.
+
+    Memory from PyArrow is not tracked at the moment.
+    """
+
+    def reset_tracking(self) -> None:
+        """Reset tracking to zero."""
+        gc.collect()
+        tracemalloc.stop()
+        tracemalloc.start()
+        assert self.get_peak() < 100_000
+
+    def get_peak(self) -> int:
+        """
+        Return peak allocated memory, in bytes.
+
+        This returns peak allocations since this object was created or
+        ``reset_tracking()`` was called, whichever is later.
+        """
+        return tracemalloc.get_traced_memory()[1]
```

Comment:
Can we add `current` as well?

Reply:
Will do.

Reply:
I did it as separate API on the theory that maybe we can figure out how to add in PyArrow eventually (though my initial attempt failed).